### PR TITLE
Expired backfills can not be updated or acknowledged

### PR DESCRIPTION
**Pull request by akremsa**
**What this PR does / Why we need it**:
Previously it was possible to "resurrect" expired backfills. With this fix, you are not allowed to do that.

### CODE REVIEW

File: internal/statestore/backfill_test.go [hunk 1]
```diff
@@ -115,57 +118,131 @@ func TestCreateBackfill(t *testing.T) {
...
 	cfg, closer := createRedis(t, false, "")
 	defer closer()
 	service := New(cfg)
 	require.NotNil(t, service)
 	defer service.Close()
 	ctx := utilTesting.NewContext(t)
 
-	bf := pb.Backfill{
-		Id:         "1",
+	// ARRANGE
```
Comment (aLekSer) [inline internal/statestore/backfill_test.go:129]: Hard to understand this comment what to Arrange?

File: internal/statestore/backfill.go [hunk 2]
```diff
@@ -208,6 +217,17 @@ func (rb *redisBackend) UpdateBackfill(ctx context.Context, backfill *pb.Backfil
 	return nil
 }
 
+func isBackfillExpired(conn redis.Conn, id string, ttl time.Duration) (bool, error) {
```
Comment (aLekSer) [inline internal/statestore/backfill.go:220]: May be it is better to remove last argument as it is not changed anywhere and the ttl duration is always has one value.

Comment (aLekSer) [inline internal/statestore/backfill.go:220]:
```suggestion
func isBackfillExpired(conn redis.Conn, id string) (bool, error) {
	ttl:=  getBackfillReleaseTimeout(rb.cfg)

```

Comment (akremsa) [reply internal/statestore/backfill.go:220]: Then I need to link this function with redisBackend struct.
I will leave it as is.

**Comment by aLekSer** [comment]
Thanks, Alexey. Code is great, but E2E tests should be fixed. When I run 20 times I see no passes, while in master E2E passes quite often:
```
for i in {0..20}; do go test -run Backfill -count 1 "open-match.dev/open-match/internal/testing/e2e" ; done >> ./test_results.log
```

**Review by aLekSer — APPROVED** [review]
Approving, code is needed to avoid Backfill zombies, when expired backfill would be not expired if it acknowledged before Cleanup find it.

**Comment by aLekSer** [comment]
One more run with 20 iterations gives me no failed runs. This is strange.

**Pull request merged**